src/structure/spatial.jl

```diff
@@ -222,6 +253,103 @@ function psiangle(res::AbstractResidue, res_next::AbstractResidue)
     return dihedralangle(res["N"], res["CA"], res["C"], res_next["N"])
 end
 
+function psiangle(chain::Chain, res_id::Union{Integer, String})
+    inds = find(r -> resid(r)==string(res_id), chain)
```

Comment:
So I played around with the code here and this is much faster:

```julia
function psiangle(chain::Chain, res_id::Union{Integer, AbstractString})
    inds = find(r -> r==string(res_id), resids(chain))
    if length(inds) != 1
        throw(ArgumentError("$res_id is an invalid residue ID"))
    end
    i = inds[1]
    res = chain[resids(chain)[i]]
    res_next = chain[resids(chain)[i+1]]
    if i == length(chain) || !sequentialresidues(res, res_next)
        throw(ArgumentError("$res_id is an invalid residue ID"))
    end
    return psiangle(res, res_next)
end
```

We use the pre-calculated lists to avoid having to `collect`. Sorry for changing my mind but I think this can just be dropped in (and edited as appropriate for the other two functions).